tutorials/basic-topics/sorting/insertion-sort.md

```diff
@@ -11,4 +11,222 @@ keywords:
+
+We have to maximize the product. Now, in order to maximize the product of two numbers we have to choose the two largest number possible. So, in order to find the maximum value of $(nums[i] - 1) * (nums[j] - 1)$ we just have to take the largest two number in the given array.
+
+Now, we know how can we maximize the value of given expression. But we have to also figure out the how can we get the two largest number present in the given array. Here comes the sorting method, if we sort our array then, the largest number would be present at the last index and second largest number would be present at the second last index of the sorted array. For sorting, we are going to use Insertion Sort Algorithm.
+
+We have figured out the solution of the problem:
+* Sort the given array using insertion sort (Refer to the algorithm section).
+* Take out the last two elements because those are the largest two elements in our array.
+* Put the values in the expression and return it.
+
+## Implementation
+
+Following are the implementation of the above mentioned problem:
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@Shivashish-rwt"/>
+
+```cpp
+class Solution {
+public:
+    void insertionSort(vector<int> &arr, int size_of_array) {
+        for (int i = 1; i < size_of_array; i++) {
+            // Picking element from array
+            int nums = arr[i]; 
+            int j = i - 1;
+            
+            // Comparing nums with all the elements left to it
+            while (j >= 0 and arr[j] > nums) { 
+                // Shifting the greater element to the right
+                arr[j + 1] = arr[j]; 
+                j--;
+            }
+            // Placing selected element at correct position
+            arr[j + 1] = nums; 
+        }
+    }
+
+    int maxProduct(vector<int>& nums) {
+        insertionSort(nums,nums.size());
+        int n = nums.size();
+        return (nums[n - 1] - 1) * (nums[n - 2] - 1);
+    }
+};
+```
+
+</TabItem>
+
+<TabItem value="java" label="Java">
+<SolutionAuthor name="@Shivashish-rwt"/>
+
+```java
+class Solution {
+    void insertion_sort(int arr[],int size_of_array) {
+		for (int i = 1; i < size_of_array; ++i) {
+                        // Picking element from array
+			int nums = arr[i]; 
+			int j = i - 1;
+
+                        // Comparing nums with all the elements left to it
+			while (j >= 0 && arr[j] > nums) { 
+                                // Shifting the greater element to the right
+				arr[j + 1] = arr[j]; 
+				j = j - 1;
+			}
+                        // Placing selected element at correct position
+			arr[j + 1] = nums; 
+		}
+	}
+    public int maxProduct(int[] nums) {
+        insertion_sort(nums,nums.length);
+        int n = nums.length;
+        return (nums[n - 1] - 1) * (nums[n - 2] - 1);      
+    }
+}
+```
+
+</TabItem>
+
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Shivashish-rwt"/>
+
+```py
+class Solution:
+    def insertionSort(arr,size_of_array):
```

Comment:
Spacing needed between method arguments.

Reply:
Check this:

![Format2](https://user-images.githubusercontent.com/85662806/194006979-010e2959-891b-4089-8b77-d2c500de1206.png)